migrator/migrations/m_189_to_m_190_vulnerability_requests_add_name/migration_impl.go

```diff
@@ -0,0 +1,198 @@
+package m189tom190
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/generated/storage"
+	newSchema "github.com/stackrox/rox/migrator/migrations/m_189_to_m_190_vulnerability_requests_add_name/schema/new"
+	"github.com/stackrox/rox/migrator/types"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/postgres/pgutils"
+	"github.com/stackrox/rox/pkg/protoconv"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/stringutils"
+	"github.com/stackrox/rox/pkg/utils"
+	"gorm.io/gorm"
+	"gorm.io/gorm/clause"
+)
+
+// Copied from central/vulnerabilityrequest/common/types.go.
+const (
+	vulnReqNameSeparator = "-"
+	defaultUserShortName = "SYS"
+)
+
+var (
+	batchSize = 2000
+	log       = logging.LoggerForModule()
+)
+
+func migrate(database *types.Databases) error {
+	ctx := sac.WithAllAccess(context.Background())
+	db := database.GormDB
+	pgutils.CreateTableFromModel(ctx, db, newSchema.CreateTableVulnerabilityRequestsStmt)
+
+	// Collect existing sequence numbers, if any.
+	lastKnownSeqNumInfo, err := collectExistingSeqNums(ctx, db)
+	if err != nil {
+		return err
+	}
+	return setRequestNames(ctx, db, lastKnownSeqNumInfo)
+}
+
+func collectExistingSeqNums(ctx context.Context, database *gorm.DB) (map[time.Month]int, error) {
+	db := database.WithContext(ctx).Table(newSchema.VulnerabilityRequestsTableName)
+	rows, err := db.Rows()
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to iterate table %s", newSchema.VulnerabilityRequestsTableName)
+	}
+	if rows.Err() != nil {
```

Comment:
This was pointed out on my PLOP prune PR, this check should go after the loop.  It has to be the last thing done for rows as rows.Err() would be populated if there were issues processing one of the batches.